pandas/core/groupby/ops.py

```diff
@@ -914,20 +916,29 @@ def agg_series(
         np.ndarray or ExtensionArray
         """
 
-        if not isinstance(obj._values, np.ndarray):
+        result = self._aggregate_series_pure_python(obj, func)
+        npvalues = lib.maybe_convert_objects(result, try_float=False)
+
+        if isinstance(obj._values, ArrowExtensionArray):
+            out = maybe_cast_pointwise_result(
+                npvalues, obj.dtype, numeric_only=True, same_dtype=False
+            )
+            import pyarrow as pa
+
+            if isinstance(out.dtype, ArrowDtype) and pa.types.is_struct(
+                out.dtype.pyarrow_dtype
+            ):
+                out = npvalues
```

Comment:
Is there a test that hits this?

Reply:
@jbrockmendel - I was surprised `maybe_cast_pointwise_result` was giving us back a Arrow dtypes we don't have EAs for.  I'm thinking the logic here to prevent this should maybe go in `dtypes.cast._maybe_cast_to_extension_array` in a followup. Any thoughts?

Reply:
> giving us back a Arrow dtypes we don't have EAs for

Can you give an example?  this confuses me.  

> should maybe go in dtypes.cast._maybe_cast_to_extension_array

_maybe_cast_to_extension_array is only used in maybe_cast_pointwise_result, so not a huge deal either way.

Reply:
```python
from pandas.core.dtypes.cast import maybe_cast_pointwise_result

arr = np.array([{"number": 1}])
result = maybe_cast_pointwise_result(
    arr, 
    dtype=pd.ArrowDtype(pa.int64()), 
    numeric_only=True, 
    same_dtype=False,
)
print(result)
# Length: 1, dtype: struct<number: int64>[pyarrow]
```

Reply:
@jbrockmendel - sorry for the noise, I was not aware we could support struct dtypes. I think everything is okay here.

@undermyumbrella1 - why go with NumPy object dtype instead of struct dtypes here?